neural_lam/models/ar_model.py

```diff
@@ -147,6 +150,42 @@ def __init__(
         # For storing spatial loss maps during evaluation
         self.spatial_loss_maps = []
 
+    def _create_dataarray_from_tensor(
+        self,
+        tensor: torch.Tensor,
+        time: Union[int, List[int]],
+        split: str,
+        category: str,
+    ) -> xr.DataArray:
+        """
+        Create an `xr.DataArray` from a tensor, with the correct dimensions and
+        coordinates to match the datastore used by the model. This function in
+        in effect is the inverse of what is returned by
+        `WeatherDataset.__getitem__`.
+
+        Parameters
+        ----------
+        tensor : torch.Tensor
+            The tensor to convert to a `xr.DataArray` with dimensions [time,
+            grid_index, feature]
+        time : Union[int,List[int]]
+            The time index or indices for the data, given as integers or a list
+            of integers representing epoch time in nanoseconds.
+        split : str
+            The split of the data, either 'train', 'val', or 'test'
+        category : str
+            The category of the data, either 'state' or 'forcing'
+        """
+        # TODO: creating an instance of WeatherDataset here on every call is
+        # not how this should be done but whether WeatherDataset should be
+        # provided to ARModel or where to put plotting still needs discussion
+        weather_dataset = WeatherDataset(datastore=self._datastore, split=split)
+        time = np.array(time, dtype="datetime64[ns]")
```

Comment:
When running on GPU `time` is a tensor that needs to be copied to CPU (tested on ohm).
`time = np.array(time.cpu(), dtype="datetime64[ns]")`

Reply:
And same for https://github.com/leifdenby/neural-lam/blob/fix/eval-vis-plots/neural_lam/weather_dataset.py#L584

`tensor.cpu().numpy()` instead of `tensor.numpy()`

Reply:
Thanks for this! I've addressed this in https://github.com/leifdenby/neural-lam/commit/85160cecf13ecfc9fc6a589ac1a9e3542da45e23